File: ios/Plugin/DetectMobileServicesPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(DetectMobileServicesPlugin, "DetectMobileServices",
           CAP_PLUGIN_METHOD(GetMediaService, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(HasHMS, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(HasGMS, CAPPluginReturnPromise);
)

File: ios/Plugin/DetectMobileServicesPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(DetectMobileServicesPlugin)
public class DetectMobileServicesPlugin: CAPPlugin {
    
    @objc func GetMediaService(_ call: CAPPluginCall) {
        call.resolve([
            "value": "ios"
        ])
    }
    @objc func HasHMS(_ call: CAPPluginCall) {
        call.resolve([
            "value": false
        ])
    }
    @objc func HasGMS(_ call: CAPPluginCall) {
        call.resolve([
            "value": false
        ])
    }
}
